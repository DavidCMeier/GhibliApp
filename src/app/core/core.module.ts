import { NgModule } from "@angular/core";
import { RootComponent } from "../root/components/root.component";
import { TranslateModule } from "@ngx-translate/core";
import { ShellComponent } from "./containers/shell/shell.component";
import { HeaderComponent } from './components/header/header.component';
import { MatToolbarModule } from "@angular/material/toolbar";
import { RouterModule } from "@angular/router";
import { MatButtonModule } from "@angular/material/button";
import { SharedModule } from "../shared/shared.module";
import { MatSelectModule } from "@angular/material/select";
import { MatOptionModule } from "@angular/material/core";
import { CommonModule } from "@angular/common";
import { FormsModule } from "@angular/forms";
import { StoreModule } from "@ngrx/store";
import * as fromState from "./store"
import { EffectsModule } from "@ngrx/effects";

@NgModule({
  declarations: [
    ShellComponent,
    HeaderComponent
  ],
  imports: [
    TranslateModule,
    MatToolbarModule,
    RouterModule,
    MatButtonModule,
    SharedModule,
    MatSelectModule,
    MatOptionModule,
    CommonModule,
    FormsModule,
    StoreModule.forFeature('core', fromState.reducers),
    EffectsModule.forFeature(fromState.effects)
  ],
  exports: [
    ShellComponent,
    HeaderComponent
  ],
  providers: [],
  bootstrap: [RootComponent]
})
export class CoreModule { }
