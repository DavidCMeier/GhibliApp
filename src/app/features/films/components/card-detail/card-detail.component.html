<div class="flex-content" *ngIf="film">
  <mat-card class="card-detail">
    <mat-card-title>{{film.title}}</mat-card-title>
    <mat-card-subtitle> {{film.director}} - {{film.release_date}}</mat-card-subtitle>
    <img mat-card-image [src]="film.movie_banner" [alt]="film.title + ' banner'">
    <mat-card-content>
      <p>{{'films.detail.original_title' | translate: {originalTitle: film.original_title} }}</p>
      <p>{{ 'films.detail.original_title_romanised' | translate:{originalTitleRomanised: film.original_title_romanised} }}</p>
      <p>{{film.description}}</p>
      <p>{{ 'films.detail.directed' | translate:{director: film.director} }}</p>
      <p>{{ 'films.detail.produced' | translate:{producer: film.producer} }}</p>
      <p>{{ 'films.detail.duration' | translate:{duration: film.running_time} }}</p>
      <p>{{ 'films.detail.rt_score' | translate:{score: film.rt_score} }}</p>
    </mat-card-content>
  </mat-card>
</div>
