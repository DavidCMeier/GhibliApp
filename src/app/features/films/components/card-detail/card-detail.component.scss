.card-detail{
  max-width: 28rem;
  box-shadow: 7px 5px 6px 0px rgb(0 0 0 / 40%);
}


