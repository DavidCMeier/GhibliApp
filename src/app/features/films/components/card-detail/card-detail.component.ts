import { Component, Input, OnInit } from '@angular/core';
import { Film } from "../../models/film.model";

@Component({
  selector: 'app-card-detail',
  templateUrl: './card-detail.component.html',
  styleUrls: ['./card-detail.component.scss']
})
export class CardDetailComponent implements OnInit {

  @Input() film!: Film;
  constructor() { }

  ngOnInit(): void {
  }

}
