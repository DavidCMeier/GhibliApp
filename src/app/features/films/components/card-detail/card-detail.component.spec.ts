import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CardDetailComponent } from './card-detail.component';
import { TranslateModule } from "@ngx-translate/core";
import { film } from "../../models/film.fixture";
import { MatCardModule } from "@angular/material/card";

describe('CardDetailComponent', () => {
  let component: CardDetailComponent;
  let fixture: ComponentFixture<CardDetailComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ CardDetailComponent ],
      imports: [TranslateModule.forRoot(), MatCardModule]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(CardDetailComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot()
  });
  it('should show one film card', () => {
    component.film = film;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot()
  });
});
