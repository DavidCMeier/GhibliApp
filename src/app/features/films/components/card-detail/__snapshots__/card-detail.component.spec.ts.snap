// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetailComponent should create 1`] = `
<app-card-detail
  __ngContext__={[Function LRootView]}
>
</app-card-detail>
`;

exports[`CardDetailComponent should show one film card 1`] = `
<app-card-detail
  __ngContext__={[Function LRootView]}
  film={[Function Object]}
>
  <div
    class="flex-content"
  >
    <mat-card
      class="mat-card mat-focus-indicator card-detail"
    >
      <mat-card-title
        class="mat-card-title"
      >
        test
      </mat-card-title>
      <mat-card-subtitle
        class="mat-card-subtitle"
      >
         test - test
      </mat-card-subtitle>
      <img
        alt="test banner"
        class="mat-card-image"
        mat-card-image=""
        src="test"
      />
      <mat-card-content
        class="mat-card-content"
      >
        <p>
          films.detail.original_title
        </p>
        <p>
          films.detail.original_title_romanised
        </p>
        <p>
          test
        </p>
        s 
        <p>
          films.detail.directed
        </p>
        <p>
          films.detail.produced
        </p>
        <p>
          films.detail.duration
        </p>
        <p>
          films.detail.rt_score
        </p>
      </mat-card-content>
    </mat-card>
  </div>
</app-card-detail>
`;
